### PR TITLE
tests: session-tool improvements

### PR DESCRIPTION
**Pull request by zyga**
There are two small patches here.
One improves working with commands with quoting
The other allows tracking the PID of started programs reliably.

### CODE REVIEW

File: tests/lib/bin/session-tool [hunk 1]
```diff
@@ -112,6 +121,18 @@ cat "$tmp_dir/ready.pipe" >/dev/null
...
+(
+	echo "#!/bin/sh"
+	test -n "$pid_file" && echo "echo \$$ >\"$pid_file\""
+	printf "exec "
+	for arg in "$@"; do
+		printf '"%s" ' "$arg"
+	done
+	echo
+)>"$tmp_dir/exec"
+# cat "$tmp_dir/exec"
```
Comment (stolowski) [inline tests/lib/bin/session-tool]: Remove?

File: tests/lib/bin/session-tool [hunk 2]
```diff
@@ -112,6 +121,18 @@ cat "$tmp_dir/ready.pipe" >/dev/null
 cat >"$tmp_dir/stdin.pipe" &
 cat <"$tmp_dir/stdout.pipe" >&1 &
 cat <"$tmp_dir/stderr.pipe" >&2 &
+(
+	echo "#!/bin/sh"
+	test -n "$pid_file" && echo "echo \$$ >\"$pid_file\""
```
Comment (bboozzoo) [inline tests/lib/bin/session-tool]: would it be possible to set a PIDFile property on the unit?

Comment (zyga) [reply tests/lib/bin/session-tool]: IMO that is not useful as the unit runs runuser, which spawns the real workload.

File: tests/lib/bin/session-tool [hunk 3]
```diff
@@ -112,6 +121,18 @@ cat "$tmp_dir/ready.pipe" >/dev/null
 cat >"$tmp_dir/stdin.pipe" &
 cat <"$tmp_dir/stdout.pipe" >&1 &
 cat <"$tmp_dir/stderr.pipe" >&2 &
+(
```
Comment (anonymouse64) [inline tests/lib/bin/session-tool]: could you add a comment that this snippet is specifically so that we can properly support quoting?

Comment (zyga) [reply tests/lib/bin/session-tool]: It's not "properly" but I get your point. (There are more bugs to fix).


I'll do a round of updates to this as I'm actively using it in another branch and add the tests.

**Comment by zyga** [comment]
This is now folded into https://github.com/snapcore/snapd/pull/8389 - with significant improvements and more tests as well.

File: tests/lib/bin/session-tool (hunk 3 above)
Comment (zyga) [reply tests/lib/bin/session-tool]: In the other branch we now have proper quote support!

File: tests/lib/bin/session-tool (hunk 1 above)
Comment (zyga) [reply tests/lib/bin/session-tool]: Removed in the follow-up to this branch.

**Pull request closed without merging**